packages/block-library/src/heading/heading-level-dropdown.js

```diff
@@ -67,36 +87,66 @@ export default function HeadingLevelDropdown( { selectedLevel, onChange } ) {
 				);
 			} }
 			renderContent={ () => (
-				<Toolbar
-					className="block-library-heading-level-toolbar"
-					__experimentalAccessibilityLabel={ __(
-						'Change heading level'
-					) }
-				>
-					<ToolbarGroup
-						isCollapsed={ false }
-						controls={ HEADING_LEVELS.map( ( targetLevel ) => {
-							const isActive = targetLevel === selectedLevel;
-							return {
-								icon: (
-									<HeadingLevelIcon
-										level={ targetLevel }
-										isPressed={ isActive }
-									/>
-								),
-								title: sprintf(
-									// translators: %s: heading level e.g: "1", "2", "3"
-									__( 'Heading %d' ),
+				<>
+					<Toolbar
+						className="block-library-heading-level-toolbar"
+						__experimentalAccessibilityLabel={ __(
+							'Change heading level'
+						) }
+					>
+						<ToolbarGroup
+							isCollapsed={ false }
+							controls={ HEADING_LEVELS.map( ( targetLevel ) => {
+								const isActive = targetLevel === selectedLevel;
+								const levelMayBeInvalid = getLevelValidity(
 									targetLevel
-								),
-								isActive,
-								onClick() {
-									onChange( targetLevel );
-								},
-							};
-						} ) }
+								).levelMayBeInvalid;
+
+								return {
+									icon: (
+										<HeadingLevelIcon
+											level={ targetLevel }
+											isPressed={ isActive }
+											isDiscouraged={ levelMayBeInvalid }
+										/>
+									),
+									title: levelMayBeInvalid
+										? sprintf(
+												// translators: %d: heading level e.g: "1", "2", "3"
+												__(
+													'Heading %d (may be invalid)'
+												),
+												targetLevel
+										  )
+										: sprintf(
+												// translators: %d: heading level e.g: "1", "2", "3"
+												__( 'Heading %d' ),
+												targetLevel
+										  ),
+									isActive,
+									onClick() {
+										onChange( targetLevel );
+									},
+								};
+							} ) }
+							aria-describedby={ `${ instanceId }__help` }
+						/>
+					</Toolbar>
+					<p
+						id={ `${ instanceId }__help` }
+						className="block-library-heading__heading-level-dropdown__help"
+					>
+						<ExternalLink href="https://www.w3.org/WAI/tutorials/page-structure/headings/">
+							{ __(
+								'Use proper heading levels to organize your content for visitors and search engines.'
+							) }
+						</ExternalLink>
+					</p>
+					<HeadingLevelWarning
```

Comment:
Maybe we can extract the display logic of `HeadingLevelWarning` here and remove the `selectedLevelMayBeInvalid` property from the component with something like this
```js
selectedLevelMayBeInvalid && <HeadingLevelWarning selectedLevel={ selectedLevel } />
```

This way it would be clear that the component always displays something, but is displayed conditionally

Reply:
Good idea. I'll do that.